storagenode/storagenodedb/database.go

```diff
@@ -115,108 +116,80 @@ func New(log *zap.Logger, config Config) (*DB, error) {
 		return nil, err
 	}
 
-	versionsPath := config.Info2
-	versionsDB, err := openDatabase(versionsPath)
-	if err != nil {
-		return nil, err
-	}
-
 	db := &DB{
 		log:    log,
 		pieces: pieces,
 		kdb:    dbs[0],
 		ndb:    dbs[1],
 		adb:    dbs[2],
 
-		// Initialize databases. Currently shares one info.db database file but
-		// in the future these will initialize their own database connections.
-		versionsDB:        newVersionsDB(versionsDB, versionsPath),
-		v0PieceInfoDB:     newV0PieceInfoDB(versionsDB, versionsPath),
-		bandwidthDB:       newBandwidthDB(versionsDB, versionsPath),
-		ordersDB:          newOrdersDB(versionsDB, versionsPath),
-		pieceExpirationDB: newPieceExpirationDB(versionsDB, versionsPath),
-		pieceSpaceUsedDB:  newPieceSpaceUsedDB(versionsDB, versionsPath),
-		reputationDB:      newReputationDB(versionsDB, versionsPath),
-		storageUsageDB:    newStorageusageDB(versionsDB, versionsPath),
-		usedSerialsDB:     newUsedSerialsDB(versionsDB, versionsPath),
-		satellitesDB:      newSatellitesDB(versionsDB, versionsPath),
+		dbDirectory: filepath.Dir(config.Info2),
+
+		sqlDatabases:      make(map[string]*sql.DB),
+		legacyInfoDB:      newLegacyInfoDB(),
+		v0PieceInfoDB:     newV0PieceInfoDB(),
+		bandwidthDB:       newBandwidthDB(),
+		ordersDB:          newOrdersDB(),
+		pieceExpirationDB: newPieceExpirationDB(),
+		pieceSpaceUsedDB:  newPieceSpaceUsedDB(),
+		reputationDB:      newReputationDB(),
+		storageUsageDB:    newStorageusageDB(),
+		usedSerialsDB:     newUsedSerialsDB(),
+		satellitesDB:      newSatellitesDB(),
 	}
 
-	return db, nil
-}
-
-// NewTest creates new test database for storage node.
-func NewTest(log *zap.Logger, storageDir string) (*DB, error) {
-	piecesDir, err := filestore.NewDir(storageDir)
+	err = db.openDatabases()
 	if err != nil {
 		return nil, err
 	}
-	pieces := filestore.New(log, piecesDir)
+	return db, nil
+}
 
-	versionsPath := ""
-	versionsDB, err := openTestDatabase()
+// openDatabases opens all the SQLite3 storage node databases and returns if any fails to open successfully.
+func (db *DB) openDatabases() error {
+	// These objeccts have a Configure method to allow setting the underlining SQLDB connection
+	// that each uses internally to do data access to the SQLite3 databases.
+	// The reason it was done this way was because there's some outside consumers that are
+	// taking a reference to the business object.
+	legacyInfoDB, err := db.openDatabase(LegacyInfoDBName)
```

Comment:
Why is it called `legacy info db`? As far as I understand we are not entirely getting rid of it, so calling it legacy doesn't seem to fit.

Reply:
I'm pretty sure we have pretty much deprecated it's use? We had a discussion about renaming this because the work @isaachess did makes migrations a per-database concept now and makes this obsolete unless I'm misunderstanding something?

Reply:
The database will exist, but will not have any data in it once all the migrations are complete. That's per the migration steps in @simongui's other PR.

Reply:
Where are versions for file migrations stored?

Reply:
I think you reviewed the approach of multi-db migrations from another PR? That PR added a versions table to each database and dynamically creates it as needed. So this database is now completely legacy and only exists for the sequence of migration steps.

We did something similar with pieceinfo when we did a big refactor there and named it `V0PieceInfo`, we named it such a way that it was clear it was older code.

If you dislike the name suggest a new one and I'll happily rename 😄

Reply:
> I think you reviewed the approach of multi-db migrations from another PR?

Not really, I saw it was too big and noticed a few problems, so it had to be split up, hence I didn't spend time doing a proper review.

But for the name. I think "deprecated" might be a slightly better choice than "legacy" because we want to indicate that we shouldn't use it anymore.

Alternatively rather than moving everything out of the main file, we could skip migrating one db out of it and repurpose the main file, so we won't have an extra file.

Reply:
Done.